adapters/eth_tx_test.go

```diff
@@ -318,3 +321,36 @@ func TestEthTxAdapter_Perform_PendingConfirmations_WithErrorInTxManager(t *testi
 
 	assert.False(t, output.HasError())
 }
+
+func TestEthTxAdapter_DeserializationBytesFormat(t *testing.T) {
+	store, cleanup := cltest.NewStore()
+	defer cleanup()
+	ctrl := gomock.NewController(t)
+	txmMock := mock_store.NewMockTxManager(ctrl)
+	store.TxManager = txmMock
+	txmMock.EXPECT().CreateTx(gomock.Any(), []byte{
+		0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x40,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x0b,
+		0x68, 0x65, 0x6c, 0x6c, 0x6f, 0x20, 0x77, 0x6f, 0x72, 0x6c, 0x64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	}).Return(&models.Tx{}, nil)
+	txmMock.EXPECT().MeetsMinConfirmations(gomock.Any())
+
+	task := models.TaskSpec{}
+	err := json.Unmarshal([]byte(`{"type": "EthTx", "params": {"format": "bytes"}}`), &task)
+	assert.NoError(t, err)
+	assert.Equal(t, task.Type, adapters.TaskTypeEthTx)
+
+	adapter, err := adapters.For(task, store)
+	assert.NoError(t, err)
+	ethtx, ok := adapter.BaseAdapter.(*adapters.EthTx)
+	assert.True(t, ok)
+	assert.Equal(t, ethtx.DataFormat, adapters.DataFormatBytes)
+
+	input := models.RunResult{
+		Data:   cltest.JSONFromString(`{"value": "hello world"}`),
+		Status: models.RunStatusInProgress,
+	}
+	result := adapter.Perform(input, store)
+	assert.Equal(t, result.Error(), "")
```

Comment:
Let's change this to
```golang
assert.False(t, result.HasError())
```

Reply:
I put this as `Equal(Error()` specifically because it shows the error when it fails, where as `assert.False(t, ... HasError` doesn't provide useful enough information to diagnose the problem.